sql/core.go

```diff
@@ -399,6 +399,38 @@ type ProjectedTable interface {
 	Projections() []string
 }
 
+// HistogramBucket represents a bucket in a histogram
+// inspiration pulled from MySQL and Cockroach DB
+type HistogramBucket struct {
+	LowerBound float64 // inclusive
+	UpperBound float64 // inclusive
+	Frequency  float64
+}
+
+// Histogram is all statistics we care about for each column
+type Histogram struct {
+	Buckets       []*HistogramBucket
+	Mean          float64
+	Min           float64
+	Max           float64
+	Count         uint64
+	NullCount     uint64
+	DistinctCount uint64
+}
+
+// HistogramMap is a map from column name to associated histogram
+type HistogramMap map[string]*Histogram
+
+// TableStatistics will hold all the statistics on a table we care about
+type TableStatistics interface {
+	CreatedAt() time.Time
+	ColumnCount() uint64
+	RowCount() uint64
```

Comment:
This is redundant with StatisticsTable.NumRows

Recommend getting rid of the latter and just using this

Reply:
There are other things things that rely on `StatisticsTable.NumRows()` like `ShowTableStatus.RowIter()`.
If those now rely on RowCount(), the table needs to be analyzed right before to provide an accurate count.

What's the best way to handle this? Make a new interface with `NumRows()`? Leave it for now?

Reply:
This isn't true. An integrator can decide to always return an accurate row count independent of when analyze is invoked.

Keep all the table stats in the same interface, no redundancies.